src/api/03-user-experience/07-autoplay.md

```diff
@@ -0,0 +1,225 @@
+  - sizes: 512x512
+    src: https://storage.googleapis.com/kino-assets/autoplay/artwork-512x512.png
+    type: image/png
+autoplay: true
+---
+
+## Introduction
+
+Sometimes the video is not just a _part_ of your web page content. Sometimes it
+is _the_ content and your users expect it to start playing automatically
+as soon as possible.
+
+There are two main ways of accomplishing this autoplay behavior:
+
+1. Using the `<video autoplay>` attribute.
+2. Using JavaScript.
+
+Let's take a look at the benefits and limitations of each.
+
+## Using the `autoplay` attribute
+
+The better known and simpler way of instructing browsers to automatically play
+a video is using the `autoplay` attribute.
+
+```html
+<video controls autoplay>
+  <source src="video.mp4" type="video/mp4">
+</video>
+```
+
+However, browsers might not honor the attribute for various reasons, likely
+because it contains sound. Automatic playback of videos with sound often leads
+to poor user experience and browsers nowadays use multiple signals to determine
+if playback with sound should be allowed.
+
+To improve your chances of browsers starting the playback, you can choose to
+play the video muted.
+
+```html
+<!-- Browsers will usually allow autoplay when
+     the `muted` attribute is also present. -->
+<video controls autoplay muted>
+  <source src="video.mp4" type="video/mp4">
+</video>
+```
+
+**Note:** Unless you provide your own UI for the video controls, make sure the
+to also use the `controls` attribute to let the browser render the default video
+controls.
+
+## Using JavaScript
+
+In JavaScript the `<video>` element contains a `play()` method that can be used
+to attempt to manually start playback as soon as possible.
+
+```html
+<video controls>
+  <source src="video.mp4" type="video/mp4">
+</video>
+<script>
+  const videoElement = document.querySelector('video');
+  videoElement.play();
+</script>
+```
+
+Similarly to the `autoplay` attribute, browsers may choose to not play the
+video in this situation, especially if it contains sound and is not muted.
+
+One useful benefit of calling the `play()` method is that it returns a Promise
+that resolves when playback starts and rejects when it won't start for any
+reason. This gives you a chance to respond.
+
+One good strategy is to mute the video and re-attempt the playback. Then, if
+that fails, render a custom play button or not do anything and let the user
+to use video controls rendered by the browser.
+
+```html
+<video controls>
+  <source src="video.mp4" type="video/mp4">
+</video>
+<script>
+  const videoElement = document.querySelector('video');
+
+  videoElement.play()
+    .catch(() => {
+      // Mute video and retry playback.
+      videoElement.muted = true;
+      videoElement.play()
+        .catch(() => {
+          // Can't autoplay with sound or muted.
+        })
+   })
+
+   /**
+    * Note: In older browsers the `play()` method may not
+    *       return a Promise. If you need to support
```

Comment:
Same here. Use `promise`

Reply:
Updated in 281290836b44cdc445dea0ef7410d1082e9d8506.